### PR TITLE
[Coverage] Enable no-unwinding-assertions in Coverage mode

### PR DESCRIPTION
**Pull request by ChenfengWei0**
The calculation of the coverage relies on the instrumented assertions. This, however, will be affected by other instrumentation like unwinding-assertions. Therefore, we need to enable `no-unwinding-assertions` by default

### CODE REVIEW

File: src/esbmc/esbmc_parseoptions.cpp [hunk 1]
```diff
@@ -1843,6 +1843,10 @@ bool esbmc_parseoptionst::process_goto_program(
       options.set_option("keep-verified-claims", false);
       options.set_option("no-pointer-check", true);
 
+      // enable '--no-unwinding-assertions' if '--unwind' is enabled
+      if (cmdline.isset("unwind"))
```
Comment (lucasccordeiro) [inline src/esbmc/esbmc_parseoptions.cpp:1847]: I think we should enable `--no-unwinding-assertion` only if the user chooses `--branch-coverage`.

@rafaelsamenezes and @mikhailramalho: What are your thoughts here?

**Review by lucasccordeiro — CHANGES_REQUESTED** [review]

File: src/esbmc/esbmc_parseoptions.cpp (hunk 1 above)
Comment (ChenfengWei0) [reply src/esbmc/esbmc_parseoptions.cpp:1847]: > I think we should enable --no-unwinding-assertion only if the user chooses --branch-coverage.

I believe that is how it works. We also apply this strategy for assertion and condition coverage as well

```
 if (
      cmdline.isset("assertion-coverage") ||
      cmdline.isset("assertion-coverage-claims"))
{
     if (cmdline.isset("unwind"))
        options.set_option("no-unwinding-assertions", true);
}
```

**Review by lucasccordeiro — APPROVED** [review]
Thanks for clarifying this, @ChenfengWei0.

**Comment by lucasccordeiro** [comment]
@ChenfengWei0: could you please add regression tests for assertion and condition coverages?

You have just added tests for branch coverages.

**Comment by lucasccordeiro** [comment]
Thanks for submitting this PR, @ChenfengWei0.

**Pull request merged**